### PR TITLE
Use advanced Bash in entrypoint

### PR DESCRIPTION
**Pull request by szepeviktor**
I know - I know - this does not look like a script.

### CODE REVIEW

File: entrypoint [hunk 1]
```diff
@@ -1,57 +1,28 @@
...
-	command_string="$command_string $action_path"
-fi
+set -e
 
-echo "Command: $command_string"
-eval "$command_string"
\ No newline at end of file
+Get_command()
+{
+	echo -n "phpstan --no-interaction --no-progress ${action_command}"
```
Comment (szepeviktor) [inline entrypoint:10]: empty action_command (and action_args and action_path below) cause an extra space when empty

**Comment by szepeviktor** [comment]
We don't have a test :(

**Comment by szepeviktor** [comment]
Please consider running PHPStan in CI.

File: entrypoint (hunk 1 above)
Comment (szepeviktor) [reply entrypoint:10]: AFAIJ only action_args is allowed to be empty...

**Comment by g105b** [comment]
Hi @szepeviktor ,

Thank you for your contribution. It looks like there are quite a few changes you're suggesting in this PR. Please could you help me understand the main benefit of the change, so I can review it later today?

Thanks,
Greg.

**Comment by szepeviktor** [comment]
I've rewritten the entrypoint from a _House of Copy-Paste_ to a single Bash function.

That is all. You can see the action options in the for loop.

**Comment by g105b** [comment]
House of Copy-Paste - that's a new one to me :smile: 

Thanks, I'm sure this will improve matters greatly. I am a fan of starting projects like this with the least code, then refactoring over time. This looks like a solid refactor, so I will merge once I've tested properly.

**Pull request merged**